File: clients/algoliasearch-client-dart/packages/client_core/lib/src/config/transformation_options.dart
import 'package:algolia_client_core/src/config/client_options.dart';

/// Options for the ingestion transporter used by `*WithTransformation` helpers.
final class TransformationOptions {
  /// The Algolia region for the Ingestion API (e.g. `'us'` or `'eu'`). Required.
  final String region;

  /// Optional overrides for the ingestion transporter's [ClientOptions].
  /// Only the fields you set here replace the Ingestion API defaults (25 s timeouts);
  /// unset fields keep those defaults.
  final ClientOptions? ingestionClientOptions;

  /// Constructs a [TransformationOptions] instance.
  TransformationOptions({
    required this.region,
    this.ingestionClientOptions,
  }) {
    if (region.isEmpty) {
      throw ArgumentError(
        'region is required in transformationOptions.'
        ' See https://www.algolia.com/doc/libraries/sdk/methods/ingestion',
      );
    }
  }

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is TransformationOptions &&
          runtimeType == other.runtimeType &&
          region == other.region &&
          ingestionClientOptions == other.ingestionClientOptions;

  @override
  int get hashCode => region.hashCode ^ ingestionClientOptions.hashCode;

  @override
  String toString() {
    return 'TransformationOptions{region: $region, ingestionClientOptions: $ingestionClientOptions}';
  }
}

File: clients/algoliasearch-client-swift/Sources/Search/Extra/TransformationOptions.swift
//
//  TransformationOptions.swift
//
//  Configuration options for the ingestion transporter used by `*WithTransformation` helpers
//  on SearchClient.
//
//  When passed to SearchClient, an ingestion transporter is created using Ingestion API
//  defaults (25 s timeouts, region-derived hosts, no compression). Pass `clientOptions` to
//  override specific defaults; any field left unset keeps the Ingestion API default.
//  Credentials are always taken from the parent SearchClient.
//
//  See https://www.algolia.com/doc/libraries/sdk/methods/ingestion

#if canImport(AlgoliaCore)
    import AlgoliaCore
#endif
import Foundation

public struct TransformationOptions {
    /// The Algolia region for the Ingestion API. Required.
    public let region: Region

    /// Optional overrides forwarded to the ingestion transporter. Fields left unset keep the
    /// Ingestion API defaults (25 s timeouts, region-derived hosts, no compression).
    public var clientOptions: ClientOptions?

    /// - parameter region: The Algolia region for the Ingestion API (`Region.us` or `Region.eu`). Required.
    /// - parameter clientOptions: Optional overrides forwarded to the ingestion transporter.
    public init(region: Region, clientOptions: ClientOptions? = nil) {
        self.region = region
        self.clientOptions = clientOptions
    }
}
